host/CMakeLists.txt

```diff
@@ -1,6 +1,38 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
+# fetch branch and last commit information of the OE SDK repository being built
+set(branch_name "Unavailable")
+set(last_commit "Unavailable")
+if (UNIX)
+execute_process(
+    COMMAND git branch
+    COMMAND grep \*
+    COMMAND cut -d " " -f2
+    COMMAND tr '\n' " "
+    OUTPUT_VARIABLE branch_name
+    RESULT_VARIABLE retcode)
+    if (NOT "${retcode}" STREQUAL "0")
+        message(WARNING "could not get branch information")
+        set(branch_name "Unavailable")
+        message(${branch_name})
+    endif()
+
+    execute_process(COMMAND git rev-parse HEAD
+                COMMAND tr '\n' " "
+                OUTPUT_VARIABLE last_commit
+                RESULT_VARIABLE retcode)
+    if (NOT "${retcode}" STREQUAL "0")
+        message(WARNING "could not to get commit information")
+        set(last_commit "Unavailable")
+        message(${branch_name})
+    endif()
+endif()
+message("branch and commit information")
+message(${branch_name})
+message(${last_commit})
+add_definitions(-DOE_REPO_BRANCH_NAME="${branch_name}" -DOE_REPO_LAST_COMMIT="${last_commit}")
```

Comment:
If these need to be set, they need to be on a target, not directory-wide. That is, please use `target_compile_definitions` instead of `add_definitions`. If there is a compelling reason to set them globally, it should be done in `compiler_settings.cmake`.

Reply:
Thanks for great suggestion.
I have replaced add_definitions with target_compile_definitions